### DISCUSSION TITLE
I created an advanced decoration plugin for wayfire: firedecor!

### DISCUSSION THREAD

**Discussion by AhoyISki** [WayfireWM/wayfire#1456] [Show and tell]
I have created an advanced plugin for decorations on wayfire, it's called [firedecor](https://github.com/AhoyISki/wayfire-firedecor), and it aims to bring very customizable windows to wayfire, here's what you can do with it right now:

* Change active and inactive colors for the borders, outline, and title.
* Change the position of decorations to any part of the layout, including the side and bottom edges.
* Add round corners to all or some of the window's corners.
* Create your own custom button theme with `png`s, or choose from three provided options.
* Set individual border sizes for each of the edges of a window.
* Create extra themes to be used on specific windows, that may change some of the options, but will keep unchanged options from the main default theme.

I feel like a good window decoration plugin was one of the things that was hampering adoption of wayfire as a window manager, but hopefully this plugin of mine is on its way to become that good window decoration.

**Comment by soreau** [comment]
Looks interesting. It is noteworthy that the wayfire decoration plugins only appear on server side decorated clients surfaces, so not an option for client side decorated client surfaces. Another thing worth mentioning is that there is [windecor plugin](https://gitlab.com/wayfireplugins/windecor) as well, which has some features you might be able to look at.

**Comment by liumingedwin** [comment]
Are you still maintaining this? It seems it is out of date!

**Comment by soreau** [comment]
There's also the [pixdecor](https://github.com/soreau/pixdecor) plugin that works with latest wayfire.

**Reply by liumingedwin** [reply]
Wow thanks!
That's what i want.
:)
